src/beacon/app/device_performance_endpoint.py

```
@@ -218,578 +405,359 @@
-    all_signal_strengths = []
     
-    current_day = start_date
-    while current_day <= end_date:
-        day_str = current_day.strftime("%Y-%m-%d")
-        day_readings = daily_readings.get(day_str, [])
+    for day_str, day_readings in daily_readings.items():
         readings_count = len(day_readings)
         
-        # Calculate daily metrics
+        # Basic performance metrics
         daily_uptime = calculate_daily_uptime(day_readings, expected_readings_per_day)
```

Ruff (0.12.2)

Undefined name `calculate_daily_uptime`

(F821)